From 4aa1baa4b6ff39fc55cc24d2c6d3306da3362502 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Fri, 2 Nov 2012 11:58:51 +0100
Subject: [PATCH] zynq: smp: Fix CONFIG_PHYS_OFFSET=0 option

The problem was when CONFIG_PHYS_OFFSET=0 because
CONFIG option was defined but this code shouldn't be called.
Fix this wrong config option.

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 arch/arm/mach-zynq/platsmp.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-zynq/platsmp.c b/arch/arm/mach-zynq/platsmp.c
index b9f007c..cbe3df7 100644
--- a/arch/arm/mach-zynq/platsmp.c
+++ b/arch/arm/mach-zynq/platsmp.c
@@ -191,7 +191,7 @@ void __init platform_smp_prepare_cpus(unsigned int max_cpus)
 	 * Remap the first three addresses at zero which are used
 	 * for 32bit long jump for SMP. Look at zynq_cpu1_start()
 	 */
-#ifdef CONFIG_PHYS_OFFSET
+#if defined(CONFIG_PHYS_OFFSET) && (CONFIG_PHYS_OFFSET != 0)
 	zero = ioremap(0, 12);
 	if (!zero) {
 		printk(KERN_WARNING
-- 
1.7.0.4

